src/cargo.rs

```diff
@@ -432,6 +434,56 @@ pub fn rustdoc_flags(config: &Config) -> String {
     value
 }
 
+fn look_for_rustflags_in(path: &Path) -> Option<String> {
+    if let Ok(contents) = read_to_string(path) {
+        let value = contents.parse::<Value>().unwrap();
```

Comment:
You can use the try operator here to return early with a `None` value by doing: `let value = contents.parse::<Value>()?;`

Reply:
I tried changing this but the compiler got very cross with me. 

It seemed to be because parse() returns an Result<F, F::Err> as opposed to an Option?

Is there an idiomatic way round this?

Reply:
ah you can use `.ok()` to turn a result into an option first so `let value = contents.parse::<Value>().ok()?;`